packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -204,22 +221,46 @@ class InMemoryAnchorService extends AnchorService {
    * @param ceramic - Ceramic API used for various purposes
    */
   set ceramic(ceramic: CeramicApi) {
-    this._ceramic = ceramic as unknown as Ceramic
-    this._dispatcher = this._ceramic.dispatcher
+    this.#ceramic = (ceramic as unknown) as Ceramic;
+    this.#dispatcher = this.#ceramic.dispatcher;
   }
 
   /**
    * Send request to the anchoring service
    * @param docId - Document ID
-   * @param cid - Commit CID
+   * @param tip - Commit CID
    */
-  async requestAnchor(docId: string, cid: CID): Promise<void> {
-    const candidate: Candidate = new Candidate(cid, docId)
-
-    if (this._anchorOnRequest) {
-      await this._process(candidate)
+  requestAnchor(docId: DocID, tip: CID): Observable<AnchorServiceResponse> {
+    const candidate = new Candidate(tip, docId);
+    const feed$ = this.#feed.pipe(
+      filter((asr) => asr.docId.equals(docId) && asr.cid.equals(tip))
+    );
+    if (this.#anchorOnRequest) {
+      requestAnchorQueue
```

Comment:
why do the anchor requests need to get serialized here?  What's wrong with letting them happen in parallel like they are currently?

Reply:
I'm still curious about this for my own understanding in case I'm missing something, but it's not a blocker, especially since this is test-only code

Reply:
Aargh, should be posted here. Valid concern. Fixed.